airflow/providers/amazon/aws/hooks/opensearch.py

```diff
@@ -0,0 +1,130 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import json
+from typing import Any
+
+from opensearchpy import AWSV4SignerAuth, OpenSearch, RequestsHttpConnection
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class OpenSearchHook(AwsBaseHook):
+    """
+    This Hook provides a thin wrapper around the OpenSearch client.
+
+    :param: open_search_conn_id: AWS Connection to use with Open Search
+    :param: log_query: Whether to log the query used for Open Search
+    """
+    conn_name_attr = "opensearch_conn_id"
+    default_conn_name = "opensearch_default"
+    conn_type = "opensearch"
+    hook_name = "AWS Open Search Hook"
+
+    def __init__(self, *args: Any, open_search_conn_id: str, log_query: bool, **kwargs: Any):
```

Comment:
```suggestion
    def __init__(self, *, open_search_conn_id: str, log_query: bool, **kwargs: Any):
```

I would recommend to accept in hook constructor keyword only arguments, positional arguments prevent to do some changes in the future without breaking changes.